internal/elasticsearch/common_test.go

```diff
@@ -564,6 +565,36 @@ func TestPodSpecHasTaintTolerations(t *testing.T) {
 	}
 }
 
+func TestElasticSearchSecurityContext(t *testing.T) {
+	podTemplate := newPodTemplateSpec(context.Background(), log.NewLogger("common-testing"), "test-node-name", "test-cluster-name", "test-namespace-name", api.ElasticsearchNode{}, api.ElasticsearchNodeSpec{}, map[string]string{}, map[api.ElasticsearchNodeRole]bool{}, nil, LogConfig{})
+
+	if !pointer.BoolEqual(podTemplate.Spec.SecurityContext.RunAsNonRoot, pointer.Bool(true)) {
+		t.Errorf("SecurityContext.RunAsNonRoot: expected true, actual %v", *podTemplate.Spec.SecurityContext.RunAsNonRoot)
+	}
+
+	expected := v1.SecurityContext{
+		AllowPrivilegeEscalation: pointer.Bool(false),
+		Capabilities: &v1.Capabilities{
+			Drop: []v1.Capability{"ALL"},
+		},
+	}
+
+	for _, c := range podTemplate.Spec.Containers {
+		if c.Name == "proxy" {
+			expected.ReadOnlyRootFilesystem = pointer.BoolPtr(true)
```

Comment:
I suggest to create two expected securityContext objects instead of re-using and partially mutating the one. Test fixtures should be as independent as possible.

Reply:
true